### PR TITLE
Kf merchants

### PR DESCRIPTION
**Pull request by TheKaterTot**
Adds merchants, customers, and transaction record endpoints. Needs to add in JBuilder still. All tests (local and test harness for these areas) passing.

### CODE REVIEW

File: app/controllers/api/v1/customers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Api::V1::CustomersController < ApplicationController
+  def index
+    render json: Customer.all.to_json(only: [:id, :first_name, :last_name])
```
Comment (TheKaterTot) [inline app/controllers/api/v1/customers_controller.rb:3]: @s-espinosa This is the first way I found to limit the returned JSON. I think I'll change it to JBuilder to format it prettier.

File: app/controllers/api/v1/customers_find_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def index
+    render json: Customer.where(find_params)
+  end
+
+  def show
+    render json: Customer.find_by(find_params)
+  end
+
+  private
+  def find_params
```
Comment (CjMoore) [inline app/controllers/api/v1/customers_find_controller.rb:12]: 👍

File: app/controllers/api/v1/customers_random_controller.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Api::V1::CustomersRandomController < ApplicationController
+  def show
+    render json: Customer.all.sample
```
Comment (CjMoore) [inline app/controllers/api/v1/customers_random_controller.rb:3]: I added a method in the application record for this, we can use it, or not.

File: db/migrate/20170314181551_change_text_to_citext.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class ChangeTextToCitext < ActiveRecord::Migration[5.0]
+  def change
+    enable_extension("citext")
+    
+    change_column :customers, :first_name, :citext
```
Comment (CjMoore) [inline db/migrate/20170314181551_change_text_to_citext.rb:5]: 👍

**Review by CjMoore — APPROVED** [review]
Looks good.

**Pull request merged**